### PR TITLE
Bump scikit requirement

### PR DESCRIPTION
**Pull request by jasonlaska**
Bump scikit-learn version and update interface to support common tests.

To do:

- update interface to support common test

### CODE REVIEW

**Comment by mnarayan** [comment]
@jasonlaska 
- Interface of some base estimator classes have changed such as the GraphLasso class and naming conventions from scikit-learn 0.19.1.  
- Might want to up our documentation of differences between us and sklearn. Support for randomized lasso has been eliminated from scikit-learn. They think it is too unreliable and think that rescaling the design matrix is equivalent to putting adaptive penalties in the regularizer. But these are not equivalent operations. I suspect the difference is related to sparsity vs. co-sparsity. So this gives our implementation an advantage.
